### ISSUE TITLE
TsTypeRef in generics doesn't pick the right type

### ISSUE THREAD

**Issue by niloc132** [Vertispan/jsinterop-ts-defs#32]
Given an "enum" type
```java
@JsType
@JsTypeDef(tsType = "string")
public class MyEnum {
  public static final String FOO = "foo";
  public static final String BAR = "bar";
}
```
it will emit ts output with an extra type for the string:
```typescript
type MyEnumType = string;
export class MyEnum {
  static readonly FOO:MyEnumType;
  static readonly BAR:MyEnumType;
}
```

It should be possible to annotate various usages of `String` throughout the project with `@TsTypeRef(MyEnum.class)` to get them to instead point to this synthetic type `MyEnumType`. For example, this String field will correctly appear to be an "enum" in ts:

```java
@TsTypeRef(MyEnum.class)
public MyEnum value;
```
```typescript
value:MyEnumType;
```

However, when using the annotation inside of generics, the "wrapper" class is picked instead

```java
public JsPropertyMap<JsArray<@TsTypeRef(MyEnum.class) JsString>> values;
```

Expected:
```typescript
values:{[key:string]:Array<MyEnumType>};
```

Actual:
```typescript
values:{[key:string]:Array<MyEnum>};
```

**Comment by niloc132** [comment]
See JsTotalsTableConfig.operationsMap and JsRollupConfig.aggregations in my d.ts branch for deephaven-core.

**Comment by niloc132** [comment]
I believe this was fixed by #33?
